File: backend/src/main/java/com/kiwi/features/quests/data/QuestDomain.java
package com.kiwi.features.quests.data;

import lombok.*;
import java.util.List;

@Getter
@Setter
public class QuestDomain {

    private int questId;
    private String name;
    private String description;
    private int experience;
    private int icon;
    private QuestStatus status;
    private List<SubquestDomain> subquests;
    private final String onCompletedEvent;
    private String onCompletedAction;
    private String onCompletedEntity;
    private final int onCompletedEntityId;

    public QuestDomain(int questId, String name, String description, int experience,
                int icon, QuestStatus status, List<SubquestDomain> subquests, String onCompletedAction, String onCompletedEntity, int onCompletedEntityId) {
        this.questId = questId;
        this.name = name;
        this.description = description;
        this.experience = experience;
        this.icon = icon;
        this.status = status;
        this.subquests = subquests;
        this.onCompletedEvent = onCompletedAction + "_" + onCompletedEntity;
        this.onCompletedAction = onCompletedAction;
        this.onCompletedEntity = onCompletedEntity;
        this.onCompletedEntityId = onCompletedEntityId;
    }
}

File: mobile/app/src/main/java/com/bellako/kiwi/features/quests/data/QuestDomain.kt
package com.bellako.kiwi.features.quests.data

data class QuestDomain(
    val id: Int,
    val name: String,
    val description: String,
    val experience: Int,
    val icon: Int,
    val status: QuestStatus,
    val subquests: List<SubquestDomain>,
    val onCompletedEvent: String,
    val onCompletedEntityId: Int,
)

enum class QuestStatus { ACTIVE, COMPLETED }
